pkg/controller/pxc/version_test.go

```diff
@@ -0,0 +1,121 @@
+package pxc
+
+import (
+	"github.com/percona/percona-xtradb-cluster-operator/pkg/apis"
+	"github.com/percona/percona-xtradb-cluster-operator/pkg/k8s"
+	"github.com/percona/percona-xtradb-cluster-operator/version"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8sruntime "k8s.io/apimachinery/pkg/runtime"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"testing"
+)
+
+func TestVersionMeta(t *testing.T) {
+	tests := []struct {
+		name            string
+		want            versionMeta
+		namespace       string
+		watchNamespaces string
+	}{
+		{
+			name: "Cluster-wide turn off",
+			want: versionMeta{
+				Apply:              "disabled",
+				ClusterWideEnabled: false,
+			},
+			namespace:       "test-namespace",
+			watchNamespaces: "test-namespace",
+		},
+		{
+			name: "Cluster-wide with specified namespaces",
+			want: versionMeta{
+				Apply:              "disabled",
+				ClusterWideEnabled: true,
+			},
+			namespace:       "test-namespace",
+			watchNamespaces: "test-namespace,another-namespace",
+		},
+		{
+			name: "Cluster-wide with empty namespaces",
+			want: versionMeta{
+				Apply:              "disabled",
+				ClusterWideEnabled: true,
+			},
+			namespace:       "test-namespace",
+			watchNamespaces: "",
+		},
+	}
+
+	size := int32(1)
+	operatorName := "percona-xtradb-cluster-operator"
+	operatorDepl := appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      operatorName,
+			Namespace: "pxc-operator",
+			Labels:    make(map[string]string),
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: &size,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"name": "percona-xtradb-cluster-operator",
+				},
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"name": "percona-xtradb-cluster-operator",
+					},
+				},
+				Spec: corev1.PodSpec{
+					ServiceAccountName: "percona-xtradb-cluster-operator",
+					Containers: []corev1.Container{
+						{
+							Name: "percona-xtradb-cluster-operator",
+						},
+					},
+				},
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+
+			t.Setenv(k8s.WatchNamespaceEnvVar, tt.watchNamespaces)
```

Comment:
Given that part of the `getVersionMeta` function's logic is to get the `WATCH_NAMESPACE` and check for an error, maybe we should configure our test to set this property variably instead of having it always being set to either empty string or a value that is valid.